src/Text/Pandoc/Readers/MediaWiki.hs

```diff
@@ -653,10 +671,15 @@ strong = B.strong <$> nested (inlinesBetween start end)
     where start = sym "'''" >> lookAhead nonspaceChar
           end   = try $ sym "'''"
 
-doubleQuotes :: MWParser Inlines
-doubleQuotes = B.doubleQuoted . trimInlines . mconcat <$> try
- ((getState >>= guard . readerSmart . mwOptions) *>
-   openDoubleQuote *> manyTill inline closeDoubleQuote )
-    where openDoubleQuote = char '"' <* lookAhead alphaNum
-          closeDoubleQuote = char '"' <* notFollowedBy alphaNum
+smart :: MWParser Inlines
+smart = do
+  getOption readerSmart >>= guard
+  doubleQuotes <|> choice [apostrophe, dash, ellipses]
```

Comment:
Heads up: For some reason, the `choice` part doesn't work (even if I remove the `doubleQuotes` part). I can open an issue once this PR gets merged.